ui/src/app/base/components/VLANSelect/VLANSelect.tsx

```diff
@@ -15,26 +18,44 @@ type Props = {
   defaultOption?: { label: string; value: string } | null;
   fabric?: VLAN["fabric"];
   includeDefaultVlan?: boolean;
+  showSpinnerOnLoad?: boolean;
+  setDefaultValueFromFabric?: boolean;
   vlans?: VLAN[] | null;
 } & FormikFieldProps;
 
 export const VLANSelect = ({
   defaultOption = { label: "Select VLAN", value: "" },
   fabric,
   includeDefaultVlan = true,
+  showSpinnerOnLoad = false,
+  setDefaultValueFromFabric,
   name,
   vlans,
+  disabled,
   ...props
 }: Props): JSX.Element => {
   const dispatch = useDispatch();
   let vlanList: VLAN[] = useSelector(vlanSelectors.all);
   const vlansLoaded = useSelector(vlanSelectors.loaded);
+  const selectedFabric = useSelector((state: RootState) =>
+    fabricSelectors.getById(state, fabric)
+  );
+  const { setFieldValue } = useFormikContext();
+
+  useEffect(() => {
+    if (setDefaultValueFromFabric) {
```

Comment:
automatically set the value to the default vlan when a new fabric is selected. Hidden behind a boolean to allow other implementations (with explicit setFieldValue on change).